src/solace_ai_connector/components/general/parser.py

```diff
@@ -8,36 +10,64 @@
 
 info = {
     "class_name": "Parser",
-    "description": "Parse a JSON string and extract data fields.",
+    "description": "Parse input from the given type to output type.",
     "config_parameters": [
         {
             "name": "input_format",
             "required": True,
-            "description": "The input format of the data. Options: 'json', 'yaml'.",
+            "description": "The input format of the data. Options: 'dict', 'json' 'yaml'. 'yaml' and 'json' must be string formatted.",
+        },
+        {
+            "name": "output_format",
+            "required": True,
+            "description": "The input format of the data. Options: 'dict', 'json' 'yaml'. 'yaml' and 'json' will be string formatted.",
         },
     ],
 }
 
+
 class Parser(ComponentBase):
     def __init__(self, **kwargs):
         super().__init__(info, **kwargs)
 
-    def invoke(self, message, data):
-        text = data["text"]
-        res_format = self.get_config("input_format", "json")
-        if res_format == "json":
-            try:
-                parser = JsonOutputParser()
-                json_res = parser.invoke(text)
-                return json_res
-            except Exception as e:
-                raise ValueError(f"Error parsing the input JSON: {str(e)}") from e
-        elif res_format == "yaml":
+    def str_to_dict(self, text, format):
+        if format == "json":
+            return JsonOutputParser().invoke(text)
+        elif format == "yaml":
             obj_text = get_obj_text("yaml", text)
-            try:
-                yaml_res = yaml.safe_load(obj_text)
-                return yaml_res
-            except Exception as e:
-                raise ValueError(f"Error parsing the input YAML: {str(e)}") from e
+            return yaml.safe_load(obj_text)
         else:
             return text
+
+    def dict_to_format(self, data, format):
+        if format == "json":
+            return json.dumps(data, indent=4)
+        elif format == "yaml":
+            return yaml.dump(data)
+        else:
+            return data
+
+    def invoke(self, message, data):
+        input_format = self.get_config("input_format")
+        output_format = self.get_config("output_format")
+        if not input_format or not output_format:
+            raise ValueError("Input and output format must be provided.")
+
+        dict_data = data  # By default assuming it's already a dictionary
+        try:
+            if input_format == "json" or input_format == "yaml":
```

Comment:
[SUG]: if it can support TXT format, support it as well rather than raising an error.